src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/FilePathNormalizer.cs

```diff
@@ -147,7 +151,7 @@ public static int GetHashCode(string filePath)
 
         foreach (var ch in normalizedSpan)
         {
-            hashCombiner.Add(ch);
+            hashCombiner.Add(charConverter(ch));
```

Comment:
A couple of suggestions:

1. On .NET Core, consider using the `StringComparer.OrdinalIgnoreCase.GetHashCode(ReadOnlySpan<char>)` or `string.GetHashCode(ReadOnlySpan<char>, StringComparison.OrdinalIgnoreCase)` overloads.
2. If converting the span to lower invariant is the way to go, consider calling `ReadOnlySpan<char>.ToLowerInvariant(...)` rather than calling into a delegate for every char.

Reply:
Thanks for this. I looked for a method on a comparer that would be useful, but didn't find one. `string.GetHashCode` will be very helpful

Reply:
It's definitely different, but for hash code I don't think we really care much, because there will still be an Equals check once the right bucket is found, and check does use ordinal comparison. In the case of hash code in a dictionary, I think its actually better to err on the side of more collisions, because we can trust Equals.